# Q&A Activity – Week 07

### Qn. 1. Carry out descriptive analytics on the dataset.
### Qn. 2. Check whether female and male students scored the same marks.
### Qn. 3. Whether test preparation helps the students

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

In [2]:
df=pd.read_csv(r"Desktop\Week_7_dataset w7.csv")
df

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

### Qn. 1. Carry out descriptive analytics on the dataset;

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


In [4]:
df.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

In [5]:
df['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

### Qn. 2. Check whether female and male students scored the same marks?

In [6]:
df1=df.groupby('Gender')
male = df.query('Gender == "male"')['Total Marks']
female = df.query('Gender == "female"')['Total Marks']
df1.describe()

Total Marks                                                      
             count       mean        std   min    25%   50%    75%   max
Gender                                                                  
female        16.0  33.750000  12.865976  12.0  25.50  39.0  43.50  50.0
male          12.0  30.416667  12.168950  14.0  21.75  29.0  43.25  47.0

In [7]:
df.groupby('Gender')['Total Marks'].mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

Hypothesis Test : Two tailed two sample t-test

In [8]:
tvalue, pvalue=stats.ttest_ind(male,female,alternative='two-sided')
print("The t value of means is: ", round(tvalue,6))
print("The p value based on the t score is: ", round(pvalue,6))

The t value of means is:  -0.694089
The p value based on the t score is:  0.493782


#### Inferences

###### p-value (0.493782) > alpha (0.05)
###### Since the p-value > alpha we fail to reject the null hypothesis (H0) & reject the alternate hypothesis (Ha).
###### Conclusion : Even though the average score of female students is higher than male students, with the help of hypothesis testing we conclude that female and male students’ marks do not significantly differ.

### Qn. 3. Whether test preparation helps the students?

In [9]:
df.groupby('Gender')['Test preparation'].value_counts()

Gender  Test preparation
female  none                12
        completed            4
male    none                 9
        completed            3
Name: Test preparation, dtype: int64

In [10]:
df.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

Hypothesis Test : One tailed two sample t-test

In [11]:
df2=df.groupby('Test preparation')
completed=df[df['Test preparation'] =="completed"]['Total Marks']
none=df[df['Test preparation'] =="none"]['Total Marks']
df2.describe()

Total Marks                                                \
                       count       mean        std   min   25%   50%   75%   
Test preparation                                                             
completed                7.0  44.571429   3.359422  40.0  43.0  44.0  46.0   
none                    21.0  28.238095  11.661495  12.0  18.0  27.0  38.0   

                        
                   max  
Test preparation        
completed         50.0  
none              48.0

In [12]:
tvalue, pvalue=stats.ttest_ind(completed,none,alternative='greater')
print("The t value of means is: ", round(tvalue,6))
print("The p value based on the t score is: ", round(pvalue,6))

The t value of means is:  3.614364
The p value based on the t score is:  0.000634


#### Inferences

###### p-value (0.000634) < alpha (0.05)
###### Since the p-value < alpha, we reject the null hypothesis (H0) & fail to reject the alternate hypothesis (Ha)
###### Conclusion : Students who completed test preparation scored significantly higher than those who did not.